projects/fal/src/fal/auth/workos_auth.py

```diff
@@ -0,0 +1,272 @@
+import functools
+import hashlib
+import os
+from base64 import urlsafe_b64encode
+from http.server import HTTPServer, SimpleHTTPRequestHandler
+from threading import Event, Thread
+from typing import Tuple, Union
+from urllib.parse import parse_qs, urlencode, urlparse
+
+import click
+import httpx
+
+from fal.auth.local import ActiveAuthConfig, UserInfo, open_browser
+from fal.console import console
+from fal.console.icons import CHECK_ICON
+
+# We need a port that's unlikely to be used by the user
+PORT = 43781
+
+WORKOS_CLIENT_ID = "client_01JCPB2MYS0GR84MZ8NJE7A3XW"
+WORKOS_DOMAIN = "api.workos.com"
+WORKOS_JWKS_URL = f"https://{WORKOS_DOMAIN}/sso/jwks/{WORKOS_CLIENT_ID}"
+WORKOS_ALGORITHMS = ["RS256"]
+WORKOS_ISSUER = "https://session.fal.ai"
+
+REDIRECT_PATH = "/api/auth/v2/callback"
+REDIRECT_URI = f"http://127.0.0.1:3000{REDIRECT_PATH}"
```

Comment:
curious about this, should this be

```suggestion
REDIRECT_PATH = "/api/auth/v2/callback"
REDIRECT_URI = f"http://127.0.0.1:{PORT}{REDIRECT_PATH}"
```